ring/ring.go

```diff
@@ -265,6 +266,9 @@ type Ring struct {
 	// to be sorted alphabetically.
 	ringZones []string
 
+	// Map containing all ring zones ever discovered, even if they have no instances.
+	allRingZones map[string]struct{}
```

Comment:
Considering the cap, I think it should be renamed:
```suggestion
	// Map containing all ring zones ever discovered, even if they have no instances,
	// capped at 32 zones to limit cardinality.
	trackedRingZones map[string]struct{}
```